File: code/03_census_geo.R
#' Download Census shapefiles
if(!require(pacman)) install.packages('pacman')
pacman::p_load(dplyr, tigris, sf, here, testthat)
setwd(here())

## Download data for mapping ---------------------------------------------------
ca_place_2010 <- places(state = 6, year = 2019)

ca_place_2010 <- ca_place_2010 |>
  st_transform(4326) |>
  transmute(place_2010_id = GEOID, place_name = NAMELSAD) |>
  arrange(place_2010_id)

ca_place_2020 <- places(state = 6)

ca_place_2020 <- ca_place_2020 |>
  st_transform(4326) |>
  transmute(place_2020_id = GEOID, place_name = NAMELSAD) |>
  arrange(place_2020_id)

cbsa <- core_based_statistical_areas(cb = TRUE)

cbsa_ca <- cbsa |>
  st_transform(4326) |>
  transmute(cbsa_2020_id = GEOID, cbsa_name = NAME) |>
  filter(grepl(", CA", cbsa_name)) |>
  arrange(cbsa_2020_id)

st_write(ca_place_2010, "data/tidy/ca_place_2010.gpkg", delete_dsn = TRUE)
st_write(ca_place_2020, "data/tidy/ca_place_2020.gpkg", delete_dsn = TRUE)
st_write(cbsa_ca, "data/tidy/ca_cbsa.gpkg", delete_dsn = TRUE)

File: code/03_census_geo.py
"""
  Download places and cbsas for California
"""

# Rewrite the commented code in python
# Path: code/03_census_geo.py

import pandas as pd
import geopandas as gpd
import numpy as np
import pygris

ca_place_2010 = pygris.places(state="06", year=2019)

ca_place_2010 = ca_place_2010.to_crs(4326)[["GEOID", "NAMELSAD", "geometry"]]\
  .rename(columns={"GEOID": "place_2010_id", "NAMELSAD": "place_name"})\
  .sort_values(by="place_2010_id")

ca_place_2020 = pygris.places(state="06")

ca_place_2020 = ca_place_2020.to_crs(4326)[["GEOID", "NAMELSAD", "geometry"]]\
  .rename(columns={"GEOID": "place_2020_id", "NAMELSAD": "place_name"})\
  .sort_values(by="place_2020_id")


cbsa = pygris.core_based_statistical_areas(cb=True)

cbsa_ca = cbsa.to_crs(4326)[["GEOID", "NAME", "geometry"]]\
  .rename(columns={"GEOID": "cbsa_2020_id", "NAME": "cbsa_name"})\
  .query("cbsa_name.str.contains(', CA')")\
  .sort_values(by="cbsa_2020_id")

ca_place_2010.to_file("data/tidy/ca_place_2010.gpkg", driver="GPKG")
ca_place_2020.to_file("data/tidy/ca_place_2020.gpkg", driver="GPKG")
cbsa_ca.to_file("data/tidy/ca_cbsa.gpkg", driver="GPKG")